pkg/controller/infinispan/resources/operatorconfig/grafana.go

```diff
@@ -0,0 +1,95 @@
+	consts "github.com/infinispan/infinispan-operator/pkg/controller/constants"
+	kube "github.com/infinispan/infinispan-operator/pkg/kubernetes"
+	grafanav1alpha1 "github.com/integr8ly/grafana-operator/v3/pkg/apis/integreatly/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	grafanaDashboardNamespace = "grafana.dashboard.namespace"
+)
+
+// reconcileGrafana reconciles grafana object status with the operator configuration settings
+func (r *ReconcileOperatorConfig) reconcileGrafana(config map[string]string, currentConfig map[string]string, operatorNs string) (*reconcile.Result, error) {
+	grafanaNs := config[grafanaDashboardNamespace]
+	currGrafanaNs := currentConfig[grafanaDashboardNamespace]
+	// Delete current grafana dashboard if namespace is changes
+	if grafanaNs != currGrafanaNs && currGrafanaNs != "" {
+		currGrafana := &grafanav1alpha1.GrafanaDashboard{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      constants.GetEnvWithDefault(ApplicationNameVariable, DefaultApplicationName),
+				Namespace: currGrafanaNs,
+			},
+		}
+		if err := r.Kube.Client.Delete(ctx, currGrafana); err != nil && !errors.IsNotFound(err) {
+			return &reconcile.Result{}, err
+		}
+	}
+	currentConfig[grafanaDashboardNamespace] = ""
+	if grafanaNs == "" {
+		return &reconcile.Result{}, nil
+	}
+	// detect the GrafanaDashboard resource type resourceExists on the cluster
+	resourceExists, err := r.Kube.IsGroupVersionSupported(grafanav1alpha1.SchemeGroupVersion.String(), grafanav1alpha1.GrafanaDashboardKind)
+	if err != nil {
+		r.Log.Error(err, "Error checking Grafana support")
+		return &reconcile.Result{Requeue: true}, nil
+	}
+	if !resourceExists {
+		r.Log.Info("Grafana CRD not present - not installing dashboard CR")
+		return &reconcile.Result{RequeueAfter: consts.DefaultWaitOnCreateResource}, nil
+	}
+	infinispanDashboard := dashboard(grafanaNs)
+	operatorDeployment := &appsv1.Deployment{}
+	result, err := kube.LookupResource("infinispan-operator", operatorNs, operatorDeployment, r.client, r.Log)
+	if err != nil {
+		r.Log.Error(err, "There was a problem looking up the infinispan operator to add the dashboard as a secondary resource")
+		return &reconcile.Result{Requeue: true}, nil
+	}
+	if result == nil && grafanaNs == operatorNs {
+		if err = controllerutil.SetControllerReference(operatorDeployment, infinispanDashboard, r.scheme); err != nil {
+			r.Log.Error(err, "Unable to set Grafana Dashboard controller reference to Infinispan operator - skipping creation")
+			return &reconcile.Result{Requeue: true}, nil
+		}
+	} else {
+		r.Log.Info("Infinispan operator resource not found, assuming operator is running locally.")
+	}
+
+	if err = r.Kube.Client.Create(ctx, infinispanDashboard); err != nil && !errors.IsAlreadyExists(err) {
```

Comment:
Can we use CreateOrUpdate in case of Grafana CR changing while parameters updates (env variable change)?